src/features/locators/components/MessageForDuplicate.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { MouseEvent } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { checkDuplicates, evaluateCssSelector, evaluateXpath } from "../utils/utils";
+import { selectLocatorsByPageObject } from "../selectors/locatorsByPO.selectors";
+import { LocatorType } from "../../../common/types/common";
+import { setActiveSingle, setScrollToLocator } from "../locators.slice";
+import { JDNHash, ElementId } from "../types/locator.types";
+
+interface MessageForDuplicateProps {
+  locator: string;
+  resetAndCloseDialog: () => void;
+  locatorType: LocatorType;
+  elementId: ElementId;
+  jdnHash: JDNHash;
+}
+
+export const MessageForDuplicate: React.FC<MessageForDuplicateProps> = ({
+  locator,
+  resetAndCloseDialog,
+  locatorType,
+  elementId,
+  jdnHash,
+}) => {
+  const dispatch = useDispatch();
+  const locators = useSelector(selectLocatorsByPageObject);
+
+  const handleOnMessageClick = async (event: MouseEvent<HTMLSpanElement>) => {
+    event.stopPropagation();
+    resetAndCloseDialog();
+
+    let locatorValue;
+    locatorType === LocatorType.cssSelector
+      ? (locatorValue = await evaluateCssSelector(locator, elementId, jdnHash))
+      : (locatorValue = await evaluateXpath(locator, elementId, jdnHash));
+
+    const { element_id: _elementId, foundHash } = JSON.parse(locatorValue);
+    const duplicates = checkDuplicates(foundHash, locators, _elementId);
+    if (duplicates.length) {
+      dispatch(setScrollToLocator(duplicates[0].element_id));
+      dispatch(setActiveSingle(duplicates[0]));
```

Comment:
what if you'll swap order of these two dispatches? will it change something now? it would be crucial for a story with empty PO, because locator doesn't appear in the list until it's not active

Reply:
It doesn't change anything, btw I changed it for correct working with empty PO case